contracts/colony/Colony.sol

```diff
@@ -109,8 +109,11 @@ contract Colony is BasicMetaTransaction, Multicall, ColonyStorage, PatriciaTreeP
       require(uint256(_amounts[i]) <= fundingPots[1].balance[token], "colony-bootstrap-not-enough-tokens");
       fundingPots[1].balance[token] = sub(fundingPots[1].balance[token], uint256(_amounts[i]));
       nonRewardPotsTotal[token] = sub(nonRewardPotsTotal[token], uint256(_amounts[i]));
+    }
 
-      assert(ERC20Extended(token).transfer(_users[i], uint256(_amounts[i])));
+    // After doing all the local storage changes, then do all the external calls
+    for (uint i = 0; i < _users.length; i++) {
```

Comment:
Is `i` being declared twice here? Stylistically I would use `j` to indicate a second loop but that's just me.

Reply:
Oh, can you declare the same identifier twice in different loops? I assumed they were in the same scope, guess not.

Reply:
Yep, they are [different scopes](https://docs.soliditylang.org/en/v0.8.17/control-structures.html#scoping-and-declarations):

> As an exception to this rule, variables declared in the initialization part of a for-loop are only visible until the end of the for-loop.